src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -91,12 +74,24 @@
 import java.util.UUID;
 
 public class SSLUtils {
-
     /**
      * The default key length to use when generating a keypair.
      */
     public static final int DEFAULT_KEY_LENGTH = 4096;
 
+    public static Class getProviderClass() throws ClassNotFoundException {
+        Class clazz;
+        try {
+            clazz = Class.forName("org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider");
+        } catch(ClassNotFoundException cnf) {
+            // if FIPS isn't present, attempt to use the non-FIPS provider. If this fails,
+            // the exception is allow to propagate
+            clazz = Class.forName("org.bouncycastle.jce.provider.BouncyCastleProvider");
+        }
+
+        return clazz;
```

Comment:
Is it possible to return the strings `"BC"` or `"BCFIPS"` (It seemed like it from the fips guides that I looked at) Are there pros and cons of that approach, if possible, vs the class based one you're doing now?

Reply:
We could, but I return the class here because I call `newInstance` on it later via reflection. See lines 1189-1191.

Reply:
Are there pros and cons to setting the provider using the class based specification vs the string based specification?

Reply:
Mainly that `builder = new JcaContentVerifierProviderBuilder().setProvider((Provider)ctor.newInstance());` won't work, as we use it elsewhere.

I adapted the existing code, which was:

```
JcaContentVerifierProviderBuilder builder =
            new JcaContentVerifierProviderBuilder().setProvider(new BouncyCastleProvider());
```

to work with either the FIPS or non-FIPS versions of BC. We used the class directly before, so I stuck to that.

Reply:
Makes sense to do the same process that was here before.

Though, it looks like it'd be a bit less involved to use the string method now that we're choosing between instances/needing to do reflection.

Reply:
I think the reflection approach is safe here though. It's constrained, and not any less fragile than a string-based approach.